js/lightbeam.js

```diff
@@ -20,6 +24,82 @@ const lightbeam = {
     });
   },
 
+  // Called from init() (isFirstParty = undefined)
+  // and redraw() (isFirstParty = true or false).
+  async updateVars(isFirstParty) {
+
+    // initialize dynamic vars from storage
+    if (!this.dataGatheredSince) {
+      const { dateStr, fullDateTime } = await this.getDataGatheredSince();
+      this.dataGatheredSince = dateStr;
+      const dataGatheredSince = document.getElementById('data-gathered-since');
+      dataGatheredSince.textContent = this.dataGatheredSince;
+      dataGatheredSince.setAttribute('datetime', fullDateTime);
+    }
+    if (isFirstParty === undefined) {
+      this.numFirstParties = await this.getNumFirstParties();
+      this.setPartyVar(true);
+      this.numThirdParties = await this.getNumThirdParties();
+      this.setPartyVar(false);
+      return;
+    }
+
+    // update on redraw
+    if (isFirstParty) {
+      this.numFirstParties++;
+      this.setPartyVar(true);
+    } else {
+      this.numThirdParties++;
+      this.setPartyVar(false);
+    }
+  },
+
+  // Updates dynamic variable values in the page
+  setPartyVar(isFirstParty) {
+    const numFirstParties = document.getElementById('num-first-parties');
+    const numThirdParties = document.getElementById('num-third-parties');
+    if (isFirstParty) {
+      if (this.numFirstParties === 0) {
+        numFirstParties.textContent = '';
+      } else {
+        numFirstParties.textContent = `${this.numFirstParties} Sites`;
+      }
+    } else if (this.numThirdParties === 0) {
+      numThirdParties.textContent = '';
+    } else {
+      const str = `${this.numThirdParties} Third Party Sites`;
+      numThirdParties.textContent = str;
+    }
+  },
+
```

Comment:
eslint is right :) I was thinking in terms of if true, else false. hehe.
i was only reading the code and didn't apply much thought. I am sorry, what you had written initially was fine.
I was seeing a pattern the way fp/tp count was set and was thinking of putting it into a function :). anyways, let me know what you think of the below comment about only returning numeric value from here and setting the text in the html page.